src/auth/googleauth.ts

```diff
@@ -594,7 +594,7 @@ export class GoogleAuth {
     }
 
     // Ensure the projectId is loaded from the keyFile if available.
-    if (this.keyFilename) {
+    if (this.keyFilename || this.jsonContent) {
```

Comment:
So I'm a little concerned here on a few fronts.  

The method is described as `getFileProjectId`, and I'm worried about concerning the scope of the method to also attempt reading from jsonContent outside the context of a file.  

The other thing I'm wondering - perhaps more importantly - will `this.jsonContent` have a `projectId` defined?  The code below will only return a `projectId` if it gets defined on the underlying `UserRefreshClient` or `JWT` objects.  If those objects are instantiated via `getClient`, it won't have a `projectId` (or at least I don't think it will).  

I don't think this is going to do what you want it to do :)

Reply:
The thing is, this method and getClient don't have the same method to determine if data can be read from jsonContent.

You don't get the same result if you call this method after calling getClient once before... That is the only think this PR tries to fix

Reply:
But maybe we should indeed just remove the if() all together since we check for the return value of getClient anyway